### ISSUE TITLE
False positive: Static methods are recognized as methods outside class interface

### ISSUE THREAD

**Issue by Ocramius** [kalessil/phpinspectionsea#1532]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 3.0.8.1 |
| Language level | e.g. PHP 7.4                        |

Given following snippet:

```php
interface A
{
    function foo() : void;
}
final class B implements A
{
    function foo() : void {}
    public static function make() : self { return new self(); }
}
```

The plugin detects that `B::make()` is not covered by interface `A`. Static method not being instance methods, this should probably not be detected:

 > [EA] Some of public methods (raise) are not part of the class contracts. Perhaps a contract is incomplete.

**Comment by kalessil** [comment]
Thank you for reporting @Ocramius: I'll look into this.

**Comment by kalessil** [comment]
Added a new setting to report static methods, disabled by default.
